scripts/api.js.php

```diff
@@ -0,0 +1,30 @@
+<?php
+$relPath="../pinc/";
+include_once($relPath."base.inc");
+
+/* This file is a JavaScript file, but to pull in variables in
+ * site_vars.php it has a PHP extension. It's important that we
+ * return the right MIME type, however, so the browser reads it
+ * as a JS file.
+ */
```

Comment:
I think you could do this by assigning the function to a variable after the page has loaded, so code_url was then available as a js_data variable. I'll try it  out to see if this makes any sense.

Reply:
I've tried it out. In the example it will work if you call makeApiAjaxSettings inside the $(function(){ ... }); and have a line in html_page_common like
```
    echo "<script>\n codeUrl = '$code_url'; </script>\n";
```
so api.js is
```javascript
/*global $ codeUrl */
/* exported makeApiAjaxSettings */

function makeApiAjaxSettings(apiUrl, queryParams) {
    let queryParamString = "";
    if(queryParams)
    {
        queryParamString = "&" + $.param(queryParams);
    }
    settings = {
        "url": codeUrl + "/api/index.php?url=" + apiUrl + queryParamString,
        "dataType": "json",
        "headers": {
            "X-API-KEY": "SESSION"
        }
    }
    return settings;
}
```
and the test function is
```javascript
$(function () {
    let settings = makeApiAjaxSettings("v1/projects/projectID45c225f598e32/pages");
    $.ajax(
        settings
    ).then(function(data) {
        console.log(data)
    });
});
```
butther could be a reason for not doing it like that.

Reply:
I like that approach better. I was trying to avoid putting JS code into every page if I could avoid it -- mostly because of some memory around it being not-ideal for use with [Content Security Policies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy) if we wanted to enable those.

I wonder, though, if we we should be instantiating a global settings object that contains a lot of what's in `site_vars.php` already. If we did that I would probably implement it similar to what I did here -- PHP masquerading as JS -- but that's also probably not as bad as this as it would be a simple object creation (probably using JSON) and less actual code. I may play with this approach a bit but will fall back to yours if mine doesn't seem like it pans out.

Reply:
After looking at the contents of `site_vars.php` I decided not to over-optimize the solution and went with your approach. If we decide we need more variables from that file exposed in JS we might move to an object later. Thanks for the suggestion!